realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1016,10 +1016,19 @@ public void executeTransaction(Transaction transaction) {
             transaction.execute(this);
             commitTransaction();
         } catch (RuntimeException e) {
```

Comment:
`error` is being caught. But the docs for Error provide the following insight: ```is the superclass of all classes that represent unrecoverable
 * errors. When errors are thrown, they should not be caught by application
 * code.```  Should this be an error? Or should this be higher order such as `Throwable`?  I'm all for throwing the root error. What is the reason for the RealmException?